CodeEdit/Features/Settings/Pages/ThemeSettings/ThemeSettingsView.swift

```diff
@@ -18,85 +18,127 @@ struct ThemeSettingsView: View {
-                            ForEach(ThemeModel.ThemeSettingsAppearances.allCases, id: \.self) { tab in
-                                Text(tab.rawValue)
-                                    .tag(tab)
+        VStack {
+            SettingsForm {
+                Section {
+                    HStack(spacing: 10) {
+                        SearchField("Search", text: $themeSearchQuery)
+
+                        Button {
+                            // As discussed, the expected behavior is to duplicate the selected theme.
+                            if let selectedTheme = themeModel.selectedTheme {
+                                if let fileURL = selectedTheme.fileURL {
+                                    themeModel.duplicate(fileURL)
+                                }
                             }
+                        } label: {
+                            Image(systemName: "plus")
+                        }
+                        .disabled(themeModel.selectedTheme == nil)
+                        .help("Create a new Theme")
+
+                        MenuWithButtonStyle(systemImage: "ellipsis", menu: {
+                            Group {
+                                Button {
+                                    themeModel.importTheme()
+                                } label: {
+                                    Text("Import Theme...")
+                                }
+                                Button {
+                                    // TODO: #1874
+                                } label: {
+                                    Text("Export All Custom Themes...")
+                                }.disabled(true)
+                            }
+                        })
+                        .padding(.horizontal, 5)
+                        .help("Import or Export Custom Themes")
+                    }
+                }
+                if themeSearchQuery.isEmpty {
+                    Section {
+                        changeThemeOnSystemAppearance
+                        if settings.matchAppearance {
+                            alwaysUseDarkTerminalAppearance
                         }
-                        .pickerStyle(.segmented)
-                        .labelsHidden()
-                        .padding(10)
+                        useThemeBackground
                     }
+                }
+                Section {
                     VStack(spacing: 0) {
-                        ForEach(
-                            themeModel.selectedAppearance == .dark
-                                ? themeModel.darkThemes
-                                : themeModel.lightThemes
-                        ) { theme in
+                        ForEach(filteredThemes) { theme in
                             Divider().padding(.horizontal, 10)
                             ThemeSettingsThemeRow(
                                 theme: $themeModel.themes[themeModel.themes.firstIndex(of: theme)!],
```

Comment:
I'm getting a crash here whenever deleting a theme while searching. For example deleting the theme here without clearing the search field.

![Screenshot 2024-09-11 at 10 43 21 PM](https://github.com/user-attachments/assets/7c7f292d-c851-454f-b7e6-088f1a0bc2f8)

Reply:
It's an unexpected `nil` error

Reply:
Good catch!